openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIHeaderMapper.java

```diff
@@ -0,0 +1,165 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.openapi.converter.service;
+
+import io.ballerina.compiler.syntax.tree.AbstractNodeFactory;
+import io.ballerina.compiler.syntax.tree.AnnotationNode;
+import io.ballerina.compiler.syntax.tree.ArrayTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.DefaultableParameterNode;
+import io.ballerina.compiler.syntax.tree.MappingConstructorExpressionNode;
+import io.ballerina.compiler.syntax.tree.MappingFieldNode;
+import io.ballerina.compiler.syntax.tree.MetadataNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.RequiredParameterNode;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SpecificFieldNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.openapi.converter.Constants;
+import io.ballerina.openapi.converter.utils.ConverterCommonUtils;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.StringSchema;
+import io.swagger.v3.oas.models.parameters.HeaderParameter;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class for the mapping ballerina headers with OAS header parameter sections.
+ *
+ * @since 2.0.0
+ */
+public class OpenAPIHeaderMapper {
+    /**
+     * Handle header parameters in ballerina data type.
+     *
+     * @param headerParam    -  {@link RequiredParameterNode} type header parameter node
+     */
+    public List<Parameter> setHeaderParameter(RequiredParameterNode headerParam) {
+        List<Parameter> parameters = new ArrayList<>();
+        //Handle with string current header a support with only string and string[]
```

Comment:
yeah we can simply remove this comment since the headers default value always be a string type. I put a comment there we directly assigned stringSchema by assuming the above reason. :)

Reply:
@lnash94 Sorry I didn't notice any changes in the latest commit. If we need to keep the comment, lets correct it.

Reply:
> The header parameters are handled separately with `optionalParameterNode` and `defaultableParameterNode`. So the `required` field will not be added for `defaltableHeaderParameter` regardless of the `treatNilableAsOptional` value right? So can we add another scenario with `treatNilableAsOptional=false` and have a default nillable header parameter? Then it is more clear. WDYT?
> 
> ```ballerina
> @http:ServiceConfig {
>      treatNilableAsOptional: false
>  }
>  service /payloadV on helloEp {
>      resource function get ping(@http:Header string? headerValue = "default") returns http:Ok {
>          http:Ok ok = {body: ()};
>          return ok;
>      }
>  }
> ```

Yeah, sure we can add this scenario too :), it is already handled below way.  For the above scenario, we don't add a required field, do we? What the expected output is 
```
/ping:
 get:
   summary: Info for a specific pet
   operationId: showPetById
   tags:
     - pets
   parameters:
     - in: header
       name: headerValue
       schema:
         type: string
         default: default
         nullable: true
```
is this correct?

Reply:
Sorry, It was my mistake, I removed comments from one overload method. I will check with HTTP module and add a proper comment.